osgtest/tests/test_29_slurm.py

```diff
@@ -88,20 +119,27 @@ def test_03_start_slurm(self):
         core.config['slurm.service-name'] = 'slurm'
         if core.el_release() == 7:
             core.config['slurm.service-name'] += 'd'
+            core.config['slurm.ctld-service-name'] = 'slurmctld'
         core.state['%s.started-service' % core.config['slurm.service-name']] = False
         self.slurm_reqs()
         self.skip_ok_if(service.is_running(core.config['slurm.service-name']), 'slurm already running')
 
         stat = core.get_stat(CTLD_LOG)
 
-        command = ['slurmctld']
-        core.check_system(command, 'enable slurmctld')
-        service.check_start(core.config['slurm.service-name'])
+        if core.el_release() == 7:
+            # slurmctld is handled by /etc/init.d/slurm on EL6
+            command = ['slurmctld']
+            core.check_system(command, 'enable slurmctld')
+            service.check_start(core.config['slurm.service-name'])
+            service.check_start(core.config['slurm.ctld-service-name'])
+        else:
+            service.check_start(core.config['slurm.service-name'])
 
         core.monitor_file(CTLD_LOG,
                           stat,
                           'slurm_rpc_node_registration complete for %s' % SHORT_HOSTNAME,
                           60.0)
+        time.sleep(10)
```

Comment:
What exactly are we waiting for here? Can we use `core.monitor_file` to look for a sentinel instead?

Reply:
When the slurm services are started, it looks like it takes them a little time to get fully operational.  The sleep it just to give the service a little time to get fully ready.  I'm not quite sure if or which bit of text gets emitted to indicate the service is ready.

Reply:
Let's try and find that sentinel instead of adding a blanket 10s wait time